sdk/eventhubs/azure_messaging_eventhubs/test-resources.bicep

```diff
@@ -0,0 +1,236 @@
+@description('Base name of the resource to be created.')
+@minLength(16)
+param baseName string = resourceGroup().name
+
+@description('The location where the resources will be created.')
+param location string = resourceGroup().location
+
+param storageEndpointSuffix string = environment().suffixes.storage
+
+param tenantIsTME bool = false
+
+@description('The client OID to grant access to test resources.')
+param testApplicationOid string
+
+var eventhubNamespaceName = 'eh-${baseName}'
+var storageAccountName = 'blb${baseName}'
+var eventHubName = 'testeventhub'
+
+resource eventhubNamespace 'Microsoft.EventHub/namespaces@2024-05-01-preview' = {
+  name: eventhubNamespaceName
+  location: location
+  sku: {
+    name: 'Standard'
+    tier: 'Standard'
+    capacity: 5
+  }
+  properties: {
+    geoDataReplication: {
+      maxReplicationLagDurationInSeconds: 0
+      locations: [
+        {
+          locationName: location
+          roleType: 'Primary'
+        }
+      ]
+    }
+    minimumTlsVersion: '1.2'
+    publicNetworkAccess: 'Enabled'
+    disableLocalAuth: !tenantIsTME // Disable local auth for TME tenants
+    zoneRedundant: false
+    isAutoInflateEnabled: false
+    maximumThroughputUnits: 0
+    kafkaEnabled: true
+  }
+}
+
+resource authorization 'Microsoft.EventHub/namespaces/authorizationrules@2024-05-01-preview' = {
+  parent: eventhubNamespace
```

Comment:
Tip: you can actually embed child resources under their parents and don't need to specify the full type name. Only caveat is that you can't refer to them by reference in, say, output variables.